### PR TITLE
Bring back MT_hashtable.h under lib/ts/

### PR DESCRIPTION
**Pull request by JordyLiu26**
Bring back MT_hashtable to lib/ts. 
The MT_hashtable have 64 locks, each lock protects a certain amount of buckets, so ideally, the HashTable was accessed by 64 threads concurrently.

### CODE REVIEW

**Comment by oknet** [comment]
[approve ci]

**Comment by oknet** [comment]
link to issue #3013

**Comment by oknet** [comment]
[approve ci clang-analyzer]

**Comment by oknet** [comment]
[approve ci clang-analyzer]

File: lib/ts/unit-tests/test_MT_hashtable.cc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+#include <cstdint>
+#include "ts/ink_platform.h"
+#include "P_EventSystem.h"
```
Comment (masaori335) [inline lib/ts/unit-tests/test_MT_hashtable.cc]: It'd be better that `MT_hashtable.h` include this file.

**Pull request merged**